### PR TITLE
feat: add `CommutativeMonoid`

### LINKED ISSUE #1681: Type Classes: Add CommutativeMonoid
Propose a design.

### PR DESCRIPTION
**Pull request by jaschdoc**
Closes #1681

I've added the `identity` law to the `Monoid` TC as that is a requirement for the algebraic structure. Should it perhaps be phrased differently like
```flix
law identity: forall(x: a) with Eq[a] . Monoid.combine(x, Monoid.empty()) == x and Monoid.combine(Monoid.empty(), x) == x
```

### CODE REVIEW

**Comment by magnus-madsen** [comment]
You could have two laws: leftIdentity and rightIdentity

File: main/src/ca/uwaterloo/flix/api/Flix.scala [hunk 1]
```diff
@@ -165,6 +165,7 @@ class Flix {
     "SemiGroup.flix" -> LocalResource.get("/src/library/SemiGroup.flix"),
     "CommutativeSemiGroup.flix" -> LocalResource.get("/src/library/CommutativeSemiGroup.flix"),
     "Monoid.flix" -> LocalResource.get("/src/library/Monoid.flix"),
+    "CommutativeMonoid.flix" -> LocalResource.get("/src/library/CommutativeMonoid.flix"),
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/api/Flix.scala:168]: This list is getting long. Can I kindly ask you to sort it alphabetically? :)

Comment (jaschdoc) [reply main/src/ca/uwaterloo/flix/api/Flix.scala:168]: Yeah sure :)

Comment (jaschdoc) [reply main/src/ca/uwaterloo/flix/api/Flix.scala:168]: Do you want the entire list sorted or just the TCs?

**Comment by jaschdoc** [comment]
Should be sorted now

**Comment by magnus-madsen** [comment]
Thanks!

**Pull request merged**